weekly-scan/template.yaml

```diff
@@ -92,6 +95,16 @@ objects:
                         } else {
                           sh "echo -e 'Weekly scan for ${image_name} was unsuccessful as image not in registry' | mail -r ${FROM_ADDRESS} -S smtp=${SMTP_SERVER} -s 'FAILED: Weekly scan ${image_name} has failed, as image not in registry' ${NOTIFY_EMAIL} "
                         }
+
+                        if (image_is_pulled == true) {
```

Comment:
Similar concern as regular builds. I'd vote for keeping this inside image removal stage.